roles/allocations/tasks/main.yml

```diff
@@ -0,0 +1,43 @@
+---
+- name: Pull allocations service container images.
+  docker_image:
+    source: pull
+    name: "{{ item }}"
+    state: present
+    force_source: yes
+  loop:
+    - "{{ allocations_balance_service_docker_image }}"
+  tags:
+    - pull
+
+- name: Configure allocations services
+  block:
+    - name: Create Docker network.
+      docker_network:
+        name: "{{ allocations_docker_network }}"
+        ipam_config:
+          - subnet: "{{ allocations_docker_network_subnet }}"
+        state: present
+      when: allocations_docker_network != "host"
+    - name: Set bind address.
```

Comment:
I recently added a Galaxy role that makes this whole process simpler. We set up all of our Docker services in more or less exactly the same manner--it made sense to generalize the pattern.

Here is an example of how it's done:

https://github.com/ChameleonCloud/chi-in-a-box/blob/421a8f0ed5c13468c677fe16f2cc7f7a57ebfb0d/roles/keycloak/tasks/main.yml#L19-L32

It simplifies the task and means you don't have to include the systemd service files at all! Want to be a beta tester? I've already used it for Keycloak but this is a good chance to try it out in a different context.